dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -0,0 +1,1078 @@
+                String pat;
+                String match_type;
+                for (size_t i = 0; i < col_size; ++i)
+                {
+                    expr_param.getStringRef(i, expr_ref);
+                    pat = pat_param.getString(i);
+                    match_type = match_type_param.getString(i);
+
+                    if (unlikely(pat.empty()))
+                        throw Exception(EMPTY_PAT_ERR_MSG);
+
+                    auto regexp = createRegexpWithMatchType(pat, match_type, collator);
+                    vec_res[i] = regexp->match(expr_ref.data, expr_ref.size); // match
+                }
+
+                res_arg.column = std::move(col_res);
+            }
+        }
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        // Do something related with nullable columns
+        NullPresence null_presence = getNullPresense(block, arguments);
+
+        const ColumnPtr & EXPR_COL_PTR_VAR_NAME = block.getByPosition(arguments[0]).column;
+
+        if (null_presence.has_null_constant)
+        {
+            block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(block.rows(), Null());
+            return;
+        }
+
+        const ColumnPtr & PAT_COL_PTR_VAR_NAME = block.getByPosition(arguments[1]).column;
+
+        size_t ARG_NUM_VAR_NAME = arguments.size();
+        auto & RES_ARG_VAR_NAME = block.getByPosition(result);
+
+        ColumnPtr MATCH_TYPE_COL_PTR_VAR_NAME;
+        if ((ARG_NUM_VAR_NAME) == 3)
+            MATCH_TYPE_COL_PTR_VAR_NAME = block.getByPosition(arguments[2]).column;
+
+        CONVERT_COLS_TO_PARAMS_AND_EXECUTE()
+    }
+
+private:
+    void checkInputArg(const DataTypePtr & arg, bool * has_nullable_col, bool * has_data_type_nothing) const
+    {
+        if (arg->isNullable())
+        {
+            *has_nullable_col = true;
+            const auto * null_type = checkAndGetDataType<DataTypeNullable>(arg.get());
+            if (null_type == nullptr)
+                throw Exception("Get unexpected nullptr in FunctionStringRegexp", ErrorCodes::LOGICAL_ERROR);
+
+            const auto & nested_type = null_type->getNestedType();
+            if (!nested_type->isString())
+            {
+                if (nested_type->getTypeId() != TypeIndex::Nothing)
+                    throw Exception(fmt::format("Illegal type {} of argument of function {}", arg->getName(), getName()), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                else
+                    *has_data_type_nothing = true;
+            }
+        }
+        else
+        {
+            if (!arg->isString())
+            {
+                if (arg->getTypeId() != TypeIndex::Nothing)
+                    throw Exception(fmt::format("Illegal type {} of argument of function {}", arg->getName(), getName()), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                else
+                    *has_data_type_nothing = true;
```

Comment:
TypeNothing should always be inside TypeNullable, so just throw exception there is ok.

Reply:
> TypeNothing should always be inside TypeNullable, so just throw exception there is ok.

okk